src/keyword.rs

```diff
@@ -117,22 +120,26 @@ impl Keyword {
             int,
             float,
             str,
+            set,
         }: OptionRaw,
     ) -> Keyword {
-        const HYPR_UNSET_FLOAT: f64 = -340282346638528859811704183484516925440.0;
-        const HYPR_UNSET_INT: i64 = -9223372036854775807;
-
-        let value = if float == HYPR_UNSET_FLOAT {
-            if int == HYPR_UNSET_INT {
-                OptionValue::String(str)
-            } else {
-                OptionValue::Int(int)
-            }
-        } else {
-            OptionValue::Float(float)
+        let int_exists = int.is_some() as u8;
+        let float_exists = float.is_some() as u8;
+        let str_exists = str.is_some() as u8;
+
+        // EXPLANATION: if at least two values is exists then we stop execution.
+        if int_exists + float_exists + str_exists > 1 {
+            panic!("The option have more than one type together! Please open an issue!")
+        }
+
+        let value = match (int, float, str) {
+            (Some(int), _, _) => OptionValue::Int(int),
+            (_, Some(float), _) => OptionValue::Float(float),
+            (_, _, Some(str)) => OptionValue::String(str),
+            _ => panic!("Unrecognized option value! Please open issue!"),
```

Comment:
This message is too vague, please make it more detailed.

Reply:
Done.